src/Npgsql/Internal/TypeHandlers/ArrayHandler.cs

```diff
@@ -297,12 +380,8 @@ public override async ValueTask<object> ReadAsObject(NpgsqlReadBuffer buf, int l
 
     #region Write
 
-    static Exception MixedTypesOrJaggedArrayException(Exception innerException)
-        => new("While trying to write an array, one of its elements failed validation. " +
-               "You may be trying to mix types in a non-generic IList, or to write a jagged array.", innerException);
-
-    static Exception CantWriteTypeException(Type type)
-        => new InvalidCastException($"Can't write type {type} as an array of {typeof(TElement)}");
+    static InvalidCastException CantWriteTypeException(Type type)
+        => new($"Can't write type {type} as an array of {typeof(TElement)}");
```

Comment:
Surprisingly, moving `CantWriteTypeException` together with `CantWriteTypeException` to base class results in 16kb size regression (no idea why, perhaps because we pass the type?).

Reply:
Odd... Maybe @NinoFloris has an idea.

Reply:
We should check after #4899 is merged.

Reply:
Yeah, still the same.